### DISCUSSION TITLE
No available kernels when using VS Code tunneling

### DISCUSSION THREAD

**Discussion by venpopov** [microsoft/vscode-jupyter#16241] [Questions and Answers]
I have setup [Remote Tunneling](https://code.visualstudio.com/docs/datascience/notebooks-web#_remote-tunneling-with-vs-code-server), and everything else works great. I can connect to my server, use the terminal, create files and run scripts in the terminal, etc. However, when I open a Jupyter notebook, I cannot connect to any of the kernels available on the remote

<img width="1408" alt="image" src="https://github.com/user-attachments/assets/fe529ada-5df9-4920-bdb2-734f0dac00c5">

I'm scratching my head as to what I'm supposed to do to be able to work with Jupyter notebooks within such a setup.

**Comment by jdunham22** [comment]
I'm having this issue as well, will try to convert the discussion to an issue

**Comment by heroheeralal01** [comment]
I was also facing the same issue, looks like you need to install the python extension, then select the python interpreter (Ctrl+Shift+P on Windows or Cmd+Shift+P on Mac).

**Comment by laxier** [comment]
i have this issue too.
